python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -316,17 +358,193 @@ def reduce(self, ratioList):
-                                                                 self.config.rejSigma)
+            self.debugRatios('reduce', ratios, ii, jj)
+        coeff, coeffErr, coeffNum = self.measureCrosstalkCoefficients(ratios, self.config.rejIter,
+                                                                      self.config.rejSigma)
         self.log.info("Coefficients:\n%s\n", coeff)
         self.log.info("Errors:\n%s\n", coeffErr)
         self.log.info("Numbers:\n%s\n", coeffNum)
         return coeff, coeffErr, coeffNum
 
-    def _getConfigName(self):
-        """Disable config output"""
-        return None
+    def measureCrosstalkCoefficients(self, ratios, rejIter=3, rejSigma=2.0):
+        """Measure crosstalk coefficients from the ratios.
 
-    def _getMetadataName(self):
-        """Disable metdata output"""
-        return None
+        Given a list of ratios for each target/source amp combination,
+        we measure a sigma clipped mean and error.
+
+        The coefficient errors returned are the standard deviation of
+        the final set of clipped input ratios.
+
+        Parameters
+        ----------
+        ratios : `list` of `list` of `numpy.ndarray`
+           Matrix of arrays of ratios.
+        rejIter : `int`
+           Number of rejection iterations.
+        rejSigma : `float`
+           Rejection threshold (sigma).
+
+        Returns
+        -------
+        coeff : `numpy.ndarray`
+            Crosstalk coefficients.
+        coeffErr : `numpy.ndarray`
+            Crosstalk coefficient errors.
+        coeffNum : `numpy.ndarray`
+            Number of pixels for each measurement.
+
+        Notes
+        -----
+        This has been moved into MeasureCrosstalkTask to allow for easier
+        debugging.
+
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['measure'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from the final set of
+            clipped input ratios.
+        """
+        if rejIter is None:
+            rejIter = self.config.rejIter
+        if rejSigma is None:
+            rejSigma = self.config.rejSigma
+
+        numAmps = len(ratios)
+        assert all(len(rr) == numAmps for rr in ratios)
+
+        coeff = np.zeros((numAmps, numAmps))
+        coeffErr = np.zeros((numAmps, numAmps))
+        coeffNum = np.zeros((numAmps, numAmps), dtype=int)
+
+        for ii, jj in itertools.product(range(numAmps), range(numAmps)):
+            if ii == jj:
+                values = [0.0]
+            else:
+                values = np.array(ratios[ii][jj])
+                values = values[np.abs(values) < 1.0]  # Discard unreasonable values
+
+            coeffNum[ii][jj] = len(values)
+
+            if len(values) == 0:
+                coeff[ii][jj] = np.nan
+                coeffErr[ii][jj] = np.nan
```

Comment:
Huh... Again, these are interesting. I get why you don't just want to set it to zero, but is the crosstalk correction code nan-safe, _i.e._ is it expecting this? I think I like this, but again, think this really might be a place to chuck in a log, probably even at WARN level, that no values remained here, so crosstalk element will be NaN.

Reply:
I've copied a similar warning from elsewhere.  I didn't originally flag this as the measured values aren't automatically used anywhere (someone needs to add them to a detector/configuration).

Reply:
Meaning this is now logging a warning? If so then mark this as resolved. I do think that logging here is appropriate, as this is the task to measure the coefficients, so it failing for any elements should be announced loudly I think.